### PR TITLE
XmppHostnameVerifier: check subject CommonName

### PR DESCRIPTION
**Pull request by huguesb**
Not all certs come with valid subjectAlternativeNames

### CODE REVIEW

File: smack-java7/src/main/java/org/jivesoftware/smack/java7/XmppHostnameVerifier.java [hunk 1]
```diff
@@ -133,7 +136,20 @@ private static void matchDns(String name, X509Certificate cert) throws Certifica
...
+            LdapName dn = new LdapName(cert.getSubjectX500Principal().getName());
+            for (Rdn rdn : dn.getRdns()) {
+                if (rdn.getType().equalsIgnoreCase("CN")) {
+                    if (matchesPerRfc2818(name, rdn.getValue().toString())) {
+                        return;
+                    }
+                    break;
+                }
+            }
+        } catch (Exception e) {
```
Comment (Flowdalic) [inline smack-java7/src/main/java/org/jivesoftware/smack/java7/XmppHostnameVerifier.java]: what does throw here?

Comment (huguesb) [inline smack-java7/src/main/java/org/jivesoftware/smack/java7/XmppHostnameVerifier.java]: LdapName ctor may throw InvalidNameException. Should the catch be more specific or the exception propagated upwards?

Comment (Flowdalic) [inline smack-java7/src/main/java/org/jivesoftware/smack/java7/XmppHostnameVerifier.java]: Ahh, I see. Please make the catch more specific and restrict the try block to only the places where the throw is expected to occur. I'm also not sure if the log level should be warning, all other logs are just 'info'. But maybe a warning is justified in this case.

**Comment by Flowdalic** [comment]
Applied as https://github.com/igniterealtime/Smack/commit/f24ef7dda918c182924f536cf2859a9fa551b575

**Pull request closed without merging**